### PR TITLE
feat(Segment): implement our endpoints using segment framework

### PR DESCRIPTION
**Pull request by kkupreeva**
### Problem
To create an integration with Segment we need to add a new destination using Segment Destinations Actions framework

### Solution
Follow this Segment documentation https://github.com/segmentio/action-destinations/blob/main/docs/create.md

- clone segment repository https://github.com/segmentio/action-destinations
- create a new destination with the name Talon.One
- create actions within this destination (trackEvent, AudienceMembershipStatusChange, Create/Delete/UpdateAduience)
- make a PR to Segment for review

### Notes
At first, I created a PR for our repository. It's just for us to look at it. After our review I will make a PR to the Segment repository and will request a review from the Segment team

### CODE REVIEW

File: .github/workflows/ext.yml [hunk 1]
```diff
@@ -2,9 +2,9 @@ name: Test External
 
 on:
   push:
-    branches:
-      - main
-  pull_request:
+#    branches:
+#      - main
+#    pull_request:
```
Comment (kkupreeva) [inline .github/workflows/ext.yml]: These changes are for checking that all tests work in the pipeline, I will remove these changes

File: packages/destination-actions/src/destinations/talon-one/__tests__/__snapshots__/snapshot.test.ts.snap [hunk 2]
```diff
@@ -0,0 +1,66 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/__tests__/__snapshots__/snapshot.test.ts.snap:1]: auto-generated file

File: packages/destination-actions/src/destinations/talon-one/__tests__/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { createTestIntegration } from '@segment/actions-core'
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/__tests__/index.test.ts:1]: tests for authorization

File: packages/destination-actions/src/destinations/talon-one/__tests__/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import Definition from '../index'
+import type { Settings } from '../generated-types'
+import nock from 'nock'
+
+const testDestination = createTestIntegration(Definition)
+
+describe('Talon One', () => {
+  describe('testAuthentication', () => {
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/__tests__/index.test.ts:9]: unit tests for authorization

File: packages/destination-actions/src/destinations/talon-one/__tests__/snapshot.test.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
+import { createTestEvent, createTestIntegration } from '@segment/actions-core'
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/__tests__/snapshot.test.ts:1]: auto-generated file (besides lines 19 and 54)

File: packages/destination-actions/src/destinations/talon-one/createAudience/__tests__/__snapshots__/snapshot.test.ts.snap [hunk 6]
```diff
@@ -0,0 +1,15 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/createAudience/__tests__/__snapshots__/snapshot.test.ts.snap:1]: auto-generated file

File: packages/destination-actions/src/destinations/talon-one/createAudience/__tests__/snapshot.test.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
+import { createTestEvent, createTestIntegration } from '@segment/actions-core'
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/createAudience/__tests__/snapshot.test.ts:1]: auto-generated file

File: packages/destination-actions/src/destinations/talon-one/deleteAudience/__tests__/__snapshots__/snapshot.test.ts.snap [hunk 8]
```diff
@@ -0,0 +1,21 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/deleteAudience/__tests__/__snapshots__/snapshot.test.ts.snap:1]: auto-generated file

File: packages/destination-actions/src/destinations/talon-one/deleteAudience/__tests__/snapshot.test.ts [hunk 9]
```diff
@@ -0,0 +1,77 @@
+import { createTestEvent, createTestIntegration } from '@segment/actions-core'
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/deleteAudience/__tests__/snapshot.test.ts:1]: auto-generated file (besides lines 19 and 53)

File: packages/destination-actions/src/destinations/talon-one/trackEvent/__tests__/__snapshots__/snapshot.test.ts.snap [hunk 10]
```diff
@@ -0,0 +1,20 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/trackEvent/__tests__/__snapshots__/snapshot.test.ts.snap:1]: auto-generated file

File: packages/destination-actions/src/destinations/talon-one/trackEvent/__tests__/snapshot.test.ts [hunk 11]
```diff
@@ -0,0 +1,75 @@
+import { createTestEvent, createTestIntegration } from '@segment/actions-core'
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/trackEvent/__tests__/snapshot.test.ts:1]: auto-generated file

File: packages/destination-actions/src/destinations/talon-one/updateAudience/__tests__/__snapshots__/snapshot.test.ts.snap [hunk 12]
```diff
@@ -0,0 +1,13 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/updateAudience/__tests__/__snapshots__/snapshot.test.ts.snap:1]: auto-generated file

File: packages/destination-actions/src/destinations/talon-one/updateAudience/__tests__/snapshot.test.ts [hunk 13]
```diff
@@ -0,0 +1,75 @@
+import { createTestEvent, createTestIntegration } from '@segment/actions-core'
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/updateAudience/__tests__/snapshot.test.ts:1]: auto-generated file

File: packages/destination-actions/src/destinations/talon-one/createAudience/index.ts [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+      label: 'audience_id',
+      description: 'Segment Audience ID (Third party audience id)',
+      type: 'string',
+      required: true
+    },
+    audience_name: {
+      label: 'audience_name',
+      description: 'Segment Audience Name (Third party audience name)',
+      type: 'string',
+      required: true
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/createAudience/index.ts]: @bolatkhanpm create audience

File: packages/destination-actions/src/destinations/talon-one/deleteAudience/index.ts [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+  description:
+    'Deletepackages/destination-actions/src/destinations/talon-one/deleteAudience/index.ts Audience in Talon.One',
+  fields: {
+    audience_id: {
+      label: 'audience_id',
+      description: 'Segment Audience ID (Third party audience id)',
+      type: 'string',
+      required: true
+    }
+  },
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/deleteAudience/index.ts]: @bolatkhanpm delete audience

File: packages/destination-actions/src/destinations/talon-one/index.ts [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+import type { Settings } from './generated-types'
+
+import createAudience from './createAudience'
+import updateAudience from './updateAudience'
+import deleteAudience from './deleteAudience'
+import trackEvent from './trackEvent'
+
+const destination: DestinationDefinition<Settings> = {
+  name: 'Talon One',
+  slug: 'talon-one',
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/index.ts]: @bolatkhanpm name and slug of our destination

File: packages/destination-actions/src/destinations/talon-one/index.ts [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+        type: 'string',
+        required: true
+      },
+      deployment: {
+        type: 'string',
+        label: 'Deployment',
+        description: 'The base URL of your Talon.One deployment.',
+        required: true
+      }
+    },
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/index.ts:29]: @bolatkhanpm 
required fields for authorization

File: packages/destination-actions/src/destinations/talon-one/trackEvent/index.ts [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+      type: 'string',
+      required: true
+    },
+    event_attributes: {
+      label: 'type',
+      description: 'Arbitrary additional JSON data associated with the event',
+      type: 'object',
+      required: false
+    }
+  },
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/trackEvent/index.ts]: @bolatkhanpm track event

File: packages/destination-actions/src/destinations/talon-one/updateAudience/index.ts [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+      type: 'string',
+      required: true
+    },
+    audience_name: {
+      label: 'audience_name',
+      description: 'Segment Audience Name (Third party audience name)',
+      type: 'string',
+      required: true
+    }
+  },
```
Comment (kkupreeva) [inline packages/destination-actions/src/destinations/talon-one/updateAudience/index.ts]: @bolatkhanpm update audience

File: packages/destination-actions/src/destinations/talon-one/createAudience/index.ts (hunk 14 above)
Comment (kkupreeva) [reply packages/destination-actions/src/destinations/talon-one/createAudience/index.ts]: `title`, `description`, `label` are fields that are displayed to users.

Comment (bolatkhanpm) [reply packages/destination-actions/src/destinations/talon-one/createAudience/index.ts]: 'You should get this Audience ID from Segment.'
'You should get this Audience Name from Segment.'

Comment (kkupreeva) [reply packages/destination-actions/src/destinations/talon-one/createAudience/index.ts]: This creates a new audience entity in Talon.One.

File: packages/destination-actions/src/destinations/talon-one/updateAudience/index.ts (hunk 19 above)
Comment (kkupreeva) [reply packages/destination-actions/src/destinations/talon-one/updateAudience/index.ts]: This synchronizes audience data if there is an existing audience entity.

File: packages/destination-actions/src/destinations/talon-one/deleteAudience/index.ts (hunk 15 above)
Comment (kkupreeva) [reply packages/destination-actions/src/destinations/talon-one/deleteAudience/index.ts]: This deletes the audience entity in Talon.One.

File: packages/destination-actions/src/destinations/talon-one/trackEvent/index.ts (hunk 18 above)
Comment (kkupreeva) [reply packages/destination-actions/src/destinations/talon-one/trackEvent/index.ts]: This records a custom event in Talon.One.
Unique identifier of the customer profile associated to the event.
It's just the name of your event.

**Review by bolatkhanpm — APPROVED** [review]
Looks good

**Pull request closed without merging**